edx-solutions/edx-platform
# Replace discussion-forum XBlock

This PR removes the old, solutions-specific `DiscussionXBlock` and replaces it with the new `DiscussionXBlock` introduced upstream. To ensure backward compatibility, the `discussion-forum` entry point provided by the old block is aliased to the `discussion` entry point provided by the new block.

The PR is part of the Eucalyptus rebase for the solutions fork.

cf. [YONK-381](https://openedx.atlassian.net/browse/YONK-381)

**Test instructions**

Assuming you are able to run Studio and LMS on the parent branch (`ziafazal/fix-broken-features`), perform these steps to check if units using the existing inline discussion block keep working after transitioning to the new block:

1.  Install the solutions version of the inline discussion block:

    1.  Add `-e common/lib/xblock/discussion` to `requirements/edx/local.txt` (otherwise the old block won't be installed):

            ...        
            -e common/lib/xblock/discussion
            -e openedx/core/lib/xblock_builtin/xblock_discussion
            -e openedx/core/lib/xblock_builtin/xblock_discussion_course

    2.  In `common/lib/xblock/discussion/setup.py`, change the package name to `discussion-forum` (to avoid a name clash with the new inline discussion block):
        
        ```python
        setup(
            name='discussion-forum',
            # ...
        )
        ```

    3.  Run `install -r requirements/edx/local.txt`.

2.  Run `pip freeze | grep discussion`. Verify that it produces the following output:
    
        discussion-edx-platform-extensions==1.0.0
        discussion-forum-0.1
        xblock-discussion==0.1
        xblock-discussion-course==0.1

3.  Add `discussion-forum` to the list of advanced modules for a course.

4.  Create a new unit and select "Advanced" > "Discussion" to add an inline discussion block. Rendering will fail because the block depends on an HTML template that has moved to another location as part of the Eucalyptus rebase. This is expected since you haven't switched to the feature branch for this PR yet.

5.  Stop Studio and uninstall the old discussion block via `pip uninstall discussion-forum`.

6.  Run `pip freeze | grep discussion`. Verify that it produces the following output:
    
        discussion-edx-platform-extensions==1.0.0
        xblock-discussion==0.1
        xblock-discussion-course==0.1

7.  Run `export NO_PREREQ_INSTALL=1` to avoid re-installing the old inline discussion block.

8.  Switch to the feature branch for this PR (`itsjeyd/replace-discussion-forum`).

9.  Run `install -r requirements/edx/local.txt` to update the new inline discussion block, which aliases the `discussion-forum` entry to `discussion`.

10. Start Studio (using `--fast` option) and reload the unit that you added a "Discussion" block to earlier. Observe that you now get a sensible preview that you can modify via the "EDIT" dialog.

**Known issues**

* In the LMS, clicking "Expand discussion" results in modal dialog saying

  > The thread you selected has been deleted. Please select another thread.

  Based on console output, there is no server-side URL for handling AJAX request from `DiscussionThreadView.prototype.loadResponses`.

  This issue is unrelated to the changes introduced in this PR (they don't touch the code that powers the new inline discussion block). A note about fixing it has been added to [YONK-401](https://openedx.atlassian.net/browse/YONK-401).
